src/handlers/cloudBackup.js

```diff
@@ -0,0 +1,120 @@
+import { sortBy } from 'lodash';
+import RNCloudFs from 'react-native-cloud-fs';
+import { RAINBOW_MASTER_KEY } from 'react-native-dotenv';
+import RNFS from 'react-native-fs';
+import AesEncryptor from '../handlers/aesEncryption';
+import { logger } from '../utils';
+const REMOTE_BACKUP_WALLET_DIR = 'rainbow.me/wallet-backups';
+const USERDATA_FILE = 'UserData.json';
+const encryptor = new AesEncryptor();
+
+export async function deleteAllBackups() {
+  const backups = await RNCloudFs.listFiles({
+    scope: 'hidden',
+    targetPath: REMOTE_BACKUP_WALLET_DIR,
+  });
+  backups.files.forEach(async file => {
+    await RNCloudFs.deleteFromCloud(file);
+  });
+}
+
+export async function encryptAndSaveDataToCloud(data, password, filename) {
+  // Encrypt the data
+  const encryptedData = await encryptor.encrypt(password, JSON.stringify(data));
+  try {
+    // Store it on the FS first
+    const path = `${RNFS.DocumentDirectoryPath}/${filename}`;
+    await RNFS.writeFile(path, encryptedData, 'utf8');
+    const sourceUri = { path };
+    const destinationPath = `${REMOTE_BACKUP_WALLET_DIR}/${filename}`;
+    const mimeType = 'application/json';
+    // Only available to our app
+    const scope = 'hidden';
+    await RNCloudFs.copyToCloud({
+      mimeType,
+      scope,
+      sourcePath: sourceUri,
+      targetPath: destinationPath,
+    });
+    // Now we need to verify the file has been stored in the cloud
+    const exists = await RNCloudFs.fileExists({
+      scope,
+      targetPath: destinationPath,
+    });
+
+    if (!exists) {
+      return false;
+    }
+
+    await RNFS.unlink(path);
+    return filename;
+  } catch (e) {
+    logger.log('Error during encryptAndSaveDataToCloud', e);
+    return false;
+  }
+}
+
+function getICloudDocument(filename) {
+  return RNCloudFs.getIcloudDocument(filename);
+}
+
+export async function getDataFromCloud(backupPassword, filename = null) {
```

Comment:
Just gonna mention: maybe we want to add some sentry logs for the failure scenarios here